src/server/pfs/server/master.go

```diff
@@ -216,9 +215,48 @@ func (d *driver) finishRepoCommits(ctx context.Context, repoKey string) error {
 				log.Error(ctx, "error finishing commit", zap.Error(err), zap.Duration("retryAfter", d))
 				return nil
 			})
-		}, zap.Bool("finishing", true), log.Proto("commit", commit), zap.String("repo", key))
+		}, zap.Bool("finishing", true), log.Proto("commit", commitInfo.Commit), zap.String("repo", key))
 	}, watch.WithSort(col.SortByCreateRevision, col.SortAscend), watch.IgnoreDelete)
-	return errors.EnsureStack(err)
+}
+
+func (d *driver) compactDiffFileset(ctx context.Context, commit *pfs.Commit, compactor *compactor, renewer *fileset.Renewer, doer task.Doer) error {
+	diffId, err := d.commitStore.GetDiffFileSet(ctx, commit)
+	if err != nil {
+		if pfsserver.IsCommitNotFoundErr(err) {
+			return nil
+		}
+		return err
+	}
+	if err := renewer.Add(ctx, *diffId); err != nil {
+		return err
+	}
+	id, err := compactor.Compact(ctx, doer, []fileset.ID{*diffId}, defaultTTL)
+	if err != nil {
+		return err
+	}
+	return errors.EnsureStack(d.commitStore.SetDiffFileSet(ctx, commit, *id))
+}
+
+func (d *driver) compactTotalFileset(ctx context.Context, commit *pfs.Commit, compactor *compactor, renewer *fileset.Renewer, doer task.Doer) (*fileset.ID, error) {
+	id, err := d.getFileSet(ctx, commit)
+	if err != nil {
+		if pfsserver.IsCommitNotFoundErr(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	if err := renewer.Add(ctx, *id); err != nil {
+		return nil, err
+	}
+	// Compact the commit. The totalID is a composed fileset that contains the diff fileset.
```

Comment:
I would remove this comment. I don't think there should be a consideration for the diff file set here, that should just be an optimization handled by the calling function. This is just a general way to compact the total file set of a commit.